### PR TITLE
don't hardcode .so, use get_shared_lib_ext instead

### PR DESCRIPTION
**Pull request by citibob**
Repeat of .dylib fix

(edit: same changes as in #788, but to `develop`)

### CODE REVIEW

File: easybuild/easyblocks/h/hdf5.py [hunk 1]
```diff
@@ -92,13 +92,14 @@ def sanity_check_step(self):
         else:
             extra_binaries = ["bin/%s" % x for x in ["h5cc", "h5fc"]]
 
+        SHLIB_EXT = get_shared_lib_ext()
```
Comment (boegel) [inline easybuild/easyblocks/h/hdf5.py]: 'constants' (i.e., variables with a name in capitals) are only defined outside of classes/functions in the coding style we maintain, so please change this to:

```python
shlib_ext = get_shared_lib_ext()
```

File: easybuild/easyblocks/h/hdf5.py [hunk 2]
```diff
@@ -92,13 +92,14 @@ def sanity_check_step(self):
...
         custom_paths = {
                         'files': ["bin/h5%s" % x for x in ["2gif", "c++", "copy", "debug", "diff",
                                                            "dump", "import", "jam","ls", "mkgrp",
                                                            "perf_serial", "redeploy", "repack",
                                                            "repart", "stat", "unjam"]] +
                                  ["bin/gif2h5"] + extra_binaries +
-                                 ["lib/libhdf5%s.so" % x for x in ["_cpp", "_fortran", "_hl_cpp",
+                                 ["lib/libhdf5%s.%s" % (x,SHLIB_EXT) for x in ["_cpp", "_fortran", "_hl_cpp",
```
Comment (boegel) [inline easybuild/easyblocks/h/hdf5.py]: nitpicking:

* space after `,`
* align the line below accordingly (see similar line with list split across multiple lines above)

File: easybuild/easyblocks/n/netcdf.py [hunk 3]
```diff
@@ -49,6 +50,8 @@ class EB_netCDF(CMakeMake):
     def configure_step(self):
         """Configure build: set config options and configure"""
 
+        SHLIB_EXT = get_shared_lib_ext()
```
Comment (boegel) [inline easybuild/easyblocks/n/netcdf.py]: please change to `shlib_ext = ...`

File: easybuild/easyblocks/n/netcdf.py [hunk 4]
```diff
@@ -74,12 +77,12 @@ def configure_step(self):
                     self.cfg.update('configopts', '-D%s_INCLUDE_DIR=%s ' % (dep.upper(), incdir))
                     if dep == 'HDF5':
                         env.setvar('HDF5_ROOT', dep_root)
-                        libhdf5 = os.path.join(dep_root, dep_libdir, 'libhdf5.so')
+                        libhdf5 = os.path.join(dep_root, dep_libdir, 'libhdf5.'+SHLIB_EXT)
```
Comment (boegel) [inline easybuild/easyblocks/n/netcdf.py]: nitpicking: space before/after `+`

Comment (boegel) [inline easybuild/easyblocks/n/netcdf.py]: also, I personally have a preference for this:

```python
'libhdf5_hl.%s' % shlib_ext
```

but that's less of an issue w.r.t. code style

**Comment by boegel** [comment]
@citibob: the proposed changes look fine, just some small remarks w.r.t. code style; maintaining a uniform coding style across the entire project really helps in lowering the bar for contributions, so please take them into account and apply the suggestions consistently

I'll also trigger our bot to run the unit tests, but I don't expect any problems there (for easyblocks, it mostly checks for silly syntax errors, it doesn't retest using the easyblock itself)

Don't hesitate to let us know if you have any questions!

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@citibob: I also tweaked the title of your PR, that's going to make it a lot easier to track (and for me to include it in the release notes for the next EB release)

**Comment by citibob** [comment]
Thank you for the input.  At this point... do I need to make the edits, or
have they already been made?  If I do need to make them, what do I need to
do on GitHub to re-submit?  This is my first pull request...

**Comment by boegel** [comment]
@citibob: no, the changes have not been made, I just added some in-line remarks in the pull request.
We're hoping that you're up for tackling the suggested changes yourself (if you're not, just let us know).

The easiest way to follow-up on them is to take a look at the 'diff' view on GitHub, which you can reach via the 'Files changed' tab (https://github.com/hpcugent/easybuild-easyblocks/pull/789/files).

That will give you an overview of the suggestions I made, alongside the lines of code to which the suggestions apply.

After making the changes, just include them in your `160101-Dylib-dev` branch in an additional commit, and push your branch to your GitHub fork of this repo again, which will update the pull request.

More or less this:

```
git checkout 160101-Dylib-dev
$EDITOR ...
git commit -a -m "fix style remarks"
git push ...
```

Don't hesitate to let us know if you have any questions!

**Comment by citibob** [comment]
OK, should be ready now for new review.  Hopefully I haven't broken
anything, I haven't re-run the builds.

More generally... I'm thinking of trying an automated search-and-replace
for the so/dylib issue.  It's really almost the only thing causing problem
on the Mac (other than package-specific Mac problems).

I figure it can't hurt to try. I'll do an automated fix, submit a pull
request, and then we can see if we like it.

-- Elizabeth

**Comment by boegel** [comment]
@citibob: no extra commit was pushed to the branch that corresponds with this PR? Do you have the commit ID?

**Comment by citibob** [comment]
Sorry, git confusion.  It should be there now.

The latest commit is SHA 27d32dea.  I "fixed" (almost) all the easyblocks.

-- Elizabeth

**Comment by boegel** [comment]
@citibob: looks great! I gave your proposed changes a thorough review, and couldn't spot any problems.

A couple of style issues came to my attention though, but since most a not caused/related to your changes, I'll tackle those in a follow-up PR, rather than block this PR over it.

Going in, thanks a lot for looking into this!

And congratulations on your first (code) contribution to EasyBuild, hopefully one of many more to come.

File: easybuild/easyblocks/a/acml.py [hunk 5]
```diff
@@ -125,7 +127,7 @@ def sanity_check_step(self):
                 inc_files.append(os.path.join(fp, 'include', 'acml%s.h' % inc))
 
             for lib in [suff] + lib_extra:
-                for ext in ['.a', '.so']:
+                for ext in ['.a', shlib_ext]:
```
Comment (boegel) [inline easybuild/easyblocks/a/acml.py]: fwiw: while testing my follow-up PR #790, I ran into a bug introduced here;

`shlib_ext` is defined as `so`, not `.so`, so I changed it to use `a` here and `.%s` in the line below, see #790

**Pull request merged**